providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/file/FileConnectorTest.java

```diff
@@ -39,12 +44,82 @@ void readAndExposeFeatureFlagsFromSource() throws IOException {
     }
 
     @Test
-    void throwsErrorIfInvalidFile(){
+    void emitErrorStateForInvalidPath() throws IOException {
         // given
         final FileConnector connector = new FileConnector("INVALID_PATH");
 
+        // when
+        connector.init();
+
+        // then
+        final BlockingQueue<StreamPayload> stream = connector.getStream();
+
+        // Must emit an error within considerable time
+        final StreamPayload[] payload = new StreamPayload[1];
+        assertTimeoutPreemptively(Duration.ofMillis(200), () -> {
+            payload[0] = stream.take();
+        });
+
+        assertNotNull(payload[0].getData());
+        assertEquals(StreamPayloadType.ERROR, payload[0].getType());
+    }
+
+    @Test
+    void watchForFileUpdatesAndEmitThem() throws IOException {
+        final String initial = "{\n" +
+                "  \"flags\": {\n" +
+                "    \"myBoolFlag\": {\n" +
+                "      \"state\": \"ENABLED\",\n" +
+                "      \"variants\": {\n" +
+                "        \"on\": true,\n" +
+                "        \"off\": false\n" +
+                "      },\n" +
+                "      \"defaultVariant\": \"on\"\n" +
+                "    }\n" +
+                "  }\n" +
+                "}";
+        final String updatedFlags = "{\n" +
+                "  \"flags\": {\n" +
+                "    \"myBoolFlag\": {\n" +
+                "      \"state\": \"ENABLED\",\n" +
+                "      \"variants\": {\n" +
+                "        \"on\": true,\n" +
+                "        \"off\": false\n" +
+                "      },\n" +
+                "      \"defaultVariant\": \"off\"\n" +
+                "    }\n" +
+                "  }\n" +
+                "}";
+
+        // given
+        final Path updPath = Paths.get(getResourcePath(UPDATABLE_FILE));
+        Files.write(updPath, initial.getBytes(), StandardOpenOption.WRITE);
```

Comment:
I would use a tmp file via [File. createTempFile()](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#createTempFile-java.lang.String-java.lang.String-) so there is no side-effects on the resources

Reply:
I already attempted this but failed when dealing with github workflow. I will try this again and revert the attempt fails

Reply:
No luck, for some reason temporary file writes within gh action CI does complete and results in timeout of this test. I will keep this test as it is for now. Beside, you can see my previous attempt to use temp files in this commit - https://github.com/open-feature/java-sdk-contrib/pull/614/commits/f548d750f7e836453b579ad85bdbb8ad8e3a2418

Reply:
Update - I disabled the file write test as it was unstable when running on the CI. However, the core functionality works fine in K8s setup and in local deployment.

Reply:
oh, I was aware of this problem in GH Action 😢 
Then, I am fine with this implementation :)

Reply:
@thisthat  I think we don't have much control over the underlying file system of the GH Action (per my search on details). Specially the write speeds and update visibility.

Reply:
Maybe is due to permissions on the job: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

Reply:
> @thisthat I think we don't have much control over the underlying file system of the GH Action (per my search on details). Specially the write speeds and update visibility.

We found this before when Sky was trying to tests file changes. We could never get this working as expected in GH actions.

Reply:
@toddbaert  😭 found it the hard way. Thank you for confirming this.